scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -0,0 +1,1226 @@
+                    if include_log_path:
+                        try:
+                            info = client.inspect_container( cid )
+                            log_path = info['LogPath'] if 'LogPath' in info else None
+                        except Exception, e:
+                          logger.error("Error inspecting container '%s'" % cid, limit_once_per_x_secs=300,limit_key="docker-api-inspect")
+
+                    result[cid] = {'name': name, 'log_path': log_path }
+
+            else:
+                result[cid] = {'name': cid, 'log_path': None}
+
+    except Exception, e:  # container querying failed
+        logger.error("Error querying running containers", limit_once_per_x_secs=300,
+                     limit_key='docker-api-running-containers' )
+        result = None
+
+    return result
+
+class PodInfo( object ):
+    """
+        A collection class that stores label and other information about a kubernetes pod
+    """
+    def __init__( self, name='', namespace='', uid='', nodeName='', labels={}, containers=[] ):
+
+        self.name = name
+        self.namespace = namespace
+        self.uid = uid
+        self.nodeName = nodeName
+        self.labels = labels
+        self.containers = containers
+
+        # generate a hash we can use to compare whether or not
+        # any of the pod info has changed
+        md5 = hashlib.md5()
+        md5.update( name )
+        md5.update( namespace )
+        md5.update( uid )
+        md5.update( nodeName )
+
+        # flatten the labels dict in to a single string
+        flattened = []
+        for k,v in labels.iteritems():
+            flattened.append( k )
+            flattened.append( v )
+        md5.update( ''.join( flattened ) )
+
+        self.digest = md5.digest()
+
+
+class ContainerChecker( StoppableThread ):
+    """
+        Monitors containers to check when they start and stop running.
+    """
+
+    def __init__( self, config, logger, socket_file, docker_api_version, host_hostname, data_path, log_path ):
+
+        self._config = config
+        self._logger = logger
+
+        self.__delay = self._config.get( 'container_check_interval' )
+        self.__log_prefix = self._config.get( 'docker_log_prefix' )
+        name = self._config.get( 'container_name' )
+
+        self.__socket_file = socket_file
+        self.__docker_api_version = docker_api_version
+        self.__client = DockerClient( base_url=('unix:/%s'%self.__socket_file), version=self.__docker_api_version )
+
+        self.container_id = self.__get_scalyr_container_id( self.__client, name )
+
+        self.__checkpoint_file = os.path.join( data_path, "docker-checkpoints.json" )
+        self.__log_path = log_path
+
+        self.__host_hostname = host_hostname
+
+        self.__readback_buffer_size = self._config.get( 'readback_buffer_size' )
+
+        self.__glob_list = config.get( 'container_globs' )
+
+        self.containers = {}
+
+        self.__k8s = Kubernetes()
```

Comment:
`Kubernetes` seems too generic of a name.  Maybe something that indicates it for the api?